generator/osm2type.cpp

```diff
@@ -834,67 +996,101 @@ void PostprocessElement(OsmElement * p, FeatureBuilderParams & params)
              params.SetReversedGeometry(true);
            }},
           {"oneway", "!", [&noOneway] { noOneway = true; }},
+// Unlike "roundabout", "circular" is not assumed to force oneway=yes
+// (https://wiki.openstreetmap.org/wiki/Tag:junction%3Dcircular), but!
+// There are a lot of junction=circular without oneway tag, which is a mapping error (run overpass under England).
+// And most of this juctions are assumed to be oneway.
```

Comment:
Typo in "juctions".
Overall its hard to understand this comment. I had to read three times :)
My understanding: in England there are many `junction=circular` which are oneway for sure (how do we know? looking at aerial images?), but they are not tagged as such, hence we force `oneway=yes` explicitly?

Reply:
Yes, we had several user claims. I checked some more "circulars" manually. They are obvious oneway (connected roads are mapped as oneway). I didn't find any circular that is not oneway. Even so, false-positive (put oneway while it is not) doesn't break routing. While false-negative makes really bad routes.